src/waveform/renderers/allshader/waveformrendererstem.cpp

```diff
@@ -46,12 +50,48 @@ bool WaveformRendererStem::init() {
     for (int stemIdx = 0; stemIdx < mixxx::kMaxSupportedStems; stemIdx++) {
         QString stemGroup = EngineDeck::getGroupForStem(m_waveformRenderer->getGroup(), stemIdx);
         m_pStemGain.emplace_back(
-                std::make_unique<PollingControlProxy>(stemGroup,
+                std::make_unique<ControlProxy>(stemGroup,
                         QStringLiteral("volume")));
+        m_pStemGain.back()->connectValueChanged(this, [this, stemIdx](double) {
+            if (!m_reorderOnChange) {
+                return;
+            }
+            m_stackOrder.removeAll(stemIdx);
+            m_stackOrder.append(stemIdx);
+        });
         m_pStemMute.emplace_back(
-                std::make_unique<PollingControlProxy>(stemGroup,
+                std::make_unique<ControlProxy>(stemGroup,
                         QStringLiteral("mute")));
+        m_pStemMute.back()->connectValueChanged(this, [this, stemIdx](double) {
+            if (!m_reorderOnChange) {
+                return;
+            }
+            m_stackOrder.removeAll(stemIdx);
+            m_stackOrder.append(stemIdx);
+        });
     }
+
```

Comment:
I'm fairly certain you can deduplicate this by simply storing the lambda in an `auto` local variable and passing it twice to `connectValueChanged`.